dj_cqrs/management/commands/cqrs_consume.py

```diff
@@ -1,50 +1,111 @@
 #  Copyright © 2022 Ingram Micro Inc. All rights reserved.
-import multiprocessing
+import logging
 import signal
+import threading
+from pathlib import Path
 
 from django.core.management.base import BaseCommand, CommandError
+from watchfiles import watch
+from watchfiles.filters import PythonFilter
+from watchfiles.run import start_process
 
 from dj_cqrs.registries import ReplicaRegistry
 from dj_cqrs.transport import current_transport
 
 
+logger = logging.getLogger('django_cqrs.cqrs_consume')
+
+
 class WorkersManager:
 
-    def __init__(self, options, transport, consume_kwargs):
+    def __init__(
+            self,
+            transport,
+            consume_kwargs,
+            workers=1,
+            reload=False,
+            ignore_paths=None,
+            sigint_timeout=5,
+            sigkill_timeout=1,
+    ):
         self.pool = []
-        self.options = options
+        self.workers = workers
+        self.reload = reload
         self.transport = transport
         self.consume_kwargs = consume_kwargs
+        self.stop_event = threading.Event()
+        self.sigint_timeout = sigint_timeout
+        self.sigkill_timeout = sigkill_timeout
+
+        if self.reload:
+            self.watch_filter = PythonFilter(ignore_paths=ignore_paths)
+            self.watcher = watch(
+                Path.cwd(),
+                watch_filter=self.watch_filter,
+                stop_event=self.stop_event,
+                yield_on_timeout=True,
+            )
+
+    def handle_signal(self, *args, **kwargs):
+        self.stop_event.set()
+
+    def run(self):
+        for sig in [signal.SIGINT, signal.SIGTERM]:
+            signal.signal(sig, self.handle_signal)
+        if self.reload:
+            signal.signal(signal.SIGHUP, self.restart)
+
+        self.start()
+
+        if self.reload:
+            for files_changed in self:
+                if files_changed:
+                    self.restart()
+        else:
+            self.stop_event.wait()
+
+        self.terminate()
 
     def start(self):
-        for i in range(self.options['workers'] or 1):
-            process = multiprocessing.Process(
-                name=f'cqrs-consumer-{i}',
-                target=self.transport.consume,
-                kwargs=self.consume_kwargs,
+        for _ in range(self.workers):
+            process = start_process(
+                self.transport.consume,
```

Comment:
The target function of `start_process` must be a function like the following:

```
def consume(**kwargs):
    import django
    django.setup()
    from dj_cqrs import transport
    transport.current_transport.consume(**kwargs)
```

otherwise in "spawn" mode (that is the default for the start_process function) it will not work.